src/main/java/picard/analysis/TheoreticalSensitivity.java

```diff
@@ -204,4 +209,193 @@ public int draw() {
+                currentDepth++;
+            }
+            // Calculate sensitivity for a particular depth, and use trapezoid rule to integrate sensitivity.
+            double left = right;
+            right = sensitivityAtConstantDepth(currentDepth, qualityHistogram, logOddsThreshold, sampleSize, alleleFraction);
+            sensitivity += deltaDepthProbability * (left + right) / 2.0;
+        }
+        return sensitivity;
+    }
+
+    /**
+     * Removes trailing zeros in a distribution.  The purpose of this function is to prevent other
+     * functions from evaluating in regions where the distribution has zero probability.
+     * @param distribution Distribution of base qualities
+     * @return Distribution of base qualities removing any trailing zeros
+     */
+     static double[] trimDistribution(final double[] distribution) {
+        int endOfDistribution = 0;
+
+        // Locate the index of the distribution where all the values remaining at
+        // larger indices are zero.
+        for (endOfDistribution = distribution.length - 1; endOfDistribution >= 0; endOfDistribution--) {
+            if (distribution[endOfDistribution] != 0) {
+                break;
+            }
+        }
+
+        // Remove trailing zeros and return.
+        return Arrays.copyOfRange(distribution, 0, endOfDistribution);
+    }
+
+    /**
+     * This is a utility function to calculate the metrics specific to running
+     * theoretical sensitivity over several different allele fractions.
+     * @param simulationSize Number of simulations to run at each depth.
+     * @param depthHistogram Histogram of depth distribution.
+     * @param baseQHistogram Histogram of Phred-scaled quality scores.
+     * @param alleleFractions List of allele fractions to measure theoretical sensitivity over.
+     */
+    public static List<TheoreticalSensitivityMetrics> calculateSensitivities(final int simulationSize,
+                                              final Histogram<Integer> depthHistogram, final Histogram<Integer> baseQHistogram, final List<Double> alleleFractions) {
+
+        final List<TheoreticalSensitivityMetrics> metricsOverVariousAlleleFractions = new ArrayList<>();
+        final double logOddsThreshold = 6.2; // This threshold is used because it is the value used for MuTect2.
+
+        // For each allele fraction in alleleFractions calculate theoretical sensitivity and add the results
+        // to the histogram sensitivityHistogram.
+        final Histogram<Double> sensitivityHistogram = new Histogram<>();
+        sensitivityHistogram.setBinLabel("allele_fraction");
+        sensitivityHistogram.setValueLabel("theoretical_sensitivity");
+        for (Double alleleFraction : alleleFractions) {
+            final TheoreticalSensitivityMetrics theoreticalSensitivityMetrics = new TheoreticalSensitivityMetrics();
+            log.info("Calculating theoretical sensitivity for AF = " + alleleFraction + ".");
```

Comment:
log.info inside this loop seems annoying. Do you expect this to take a long time?